code/go/pkg/validator/validator_test.go

```diff
@@ -722,7 +715,72 @@ func TestValidateRoutingRules(t *testing.T) {
 			}
 		})
 	}
+}
+
+func TestValidateIngestPipelines(t *testing.T) {
+	tests := map[string]map[string][]string{
+		"good":    {},
+		"good_v2": {},
+		// "good_v3": {}, // TODO: this fails with "package with GA version (1.0.0) is using an unreleased version of the spec (3.1.0-next)"
+		"skip_pipeline_rename_validation": {},
+		"bad_ingest_pipeline": {
+			"test": []string{
+				"field processors.1: Additional property reroute is not allowed",
+				"field processors.2.foreach.processor: Additional property paint is not allowed",
+			},
+			"bad_rename_message": []string{
+				"field processors.1.rename: if is required (JSE00001)",
+				"field processors.0: remove is required (JSE00001)",
+			},
+			"bad_rename_message_2": []string{
+				"field processors.2.remove.field: processors.2.remove.field does not match: \"message\" (JSE00001)",
+				"field processors.2.remove.if: processors.2.remove.if does not match: \"ctx.event?.original != null\" (JSE00001)",
+			},
```

Comment:
Would it make sense to replace these error messages, so they are more significant for this use case?

Something like this:
```suggestion
			"bad_rename_message": []string{
				"field processors.1.rename: at least one is required to move \"message\" to \"event.original\" (JSE00001)",
				"field processors.0: remove is required to remove \"message\" when \"event.original\" exists (JSE00001)",
			},
			"bad_rename_message_2": []string{
				"field processors.2.remove.field: processors.2.remove.field at least one remove processor should remove \"message\" (JSE00001)",
				"field processors.2.remove.if: processors.2.remove.if does not match: \"ctx.event?.original != null\" (JSE00001)",
			},
```

My only concern would be that maybe we replace the message for other cases, but I don't think we are enforcing these conditions in other cases.

Reply:
Good point 👍 This should be fairly safe as long as we don't add other similar conditions. I can add a comment in the code that these renamed error messages should be checked in case this changes.

Reply:
I pushed a commit with renamed error messages, let me know what you think. I made the rules as strict as possible to avoid collisions with other cases. For example, just using `if is required` would also have affected [this test](https://github.com/elastic/package-spec/blob/b8224e23383208ce3688c1e63b1a4795b1037f3f/code/go/pkg/validator/validator_test.go#L699).

Reply:
Actually, looking at the integrations tests, I realised that we can't hardcode the processor number so I would need to change that to a regex.

Reply:
Oh I see. What do you think about adding support for regexps in the transforms?

Reply:
Pushed a commit with that, let's see how it impact the integrations.